src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main.go

```diff
@@ -63,33 +78,49 @@ func (cfg *Config) Setup() (cmd.Starter, error) {
 		flags.StringVar(&tlsConfig.KeyFile, "cert-key", "", "cert key")
 		flags.BoolVar(&tlsConfig.SkipVerify, "tls-skip-verify", false, "Skip TLS certificate verification - use it only for testing")
 		flags.BoolVar(&disableTLS, "disable-tls", false, "Disable TLS encryption - use it only for testing")
+		flags.BoolVar(&secureComms, "secure-comms", false, "Use SSH to secure communication between cluster and peer pods")
+		flags.StringVar(&secureCommsInbounds, "secure-comms-inbounds", "", "Inbound tags for secure communication tunnels")
+		flags.StringVar(&secureCommsOutbounds, "secure-comms-outbounds", "", "Outbound tags for secure communication tunnels")
 	})
 
-	if !disableTLS {
-		cfg.tlsConfig = &tlsConfig
-	}
-
 	cmd.ShowVersion(programName)
 
 	if showVersion {
 		cmd.Exit(0)
 	}
 
-	for path, obj := range map[string]interface{}{
-		cfg.configPath: &cfg.daemonConfig,
-	} {
-		if err := load(path, obj); err != nil {
-			return nil, err
+	if err := load(cfg.configPath, &cfg.daemonConfig); err != nil {
+		return nil, err
+	}
+
+	if secureComms {
+		ppssh.Singleton()
+		host, port, err := net.SplitHostPort(cfg.listenAddr)
+		if err != nil {
+			logger.Printf("Warning: illegal address %s is changed to 127.0.0.1:15150 since secure-comms is enabled.", cfg.listenAddr)
+		}
+		if host != "127.0.0.1" {
+			logger.Printf("Address %s is changed to 127.0.0.1:%s since secure-comms is enabled.", cfg.listenAddr, port)
+			cfg.listenAddr = "127.0.0.1:" + port
+		}
+		inbounds := append([]string{"B:KBS:8080"}, strings.Split(secureCommsInbounds, ",")...)
+		outbounds := append([]string{"K:KATAAPI:" + port}, strings.Split(secureCommsOutbounds, ",")...)
+
+		apic := apic.NewApiClient(API_SERVER_REST_PORT, cfg.kataAgentNamespace)
```

Comment:
A comment explaining why the following approach of setting the method name ie, apic.GetKey for GetSecret
was chosen. Additionally also a short comment on the flow ppssh->ppsecrets->apic
This will help in understanding the code better for anyone looking at it later on

Reply:
Added a comment

Reply:
nice.. nit: s/than/then